app.go

```diff
@@ -154,12 +156,44 @@ func New(config Config) (*App, error) {
 		wg:         sync.WaitGroup{},
 	}
 
-	err := app.connect("")
-	if err != nil {
-		log.Logger.Errorf("Failed to connect the app: %v", err)
-		return nil, err
-	}
+	app.ctx, app.ctxCancel = context.WithCancel(context.Background())
+
+	app.wg.Add(1)
+	go func() {
+		defer app.wg.Done()
+		var reconnectDelay int64 = 1
```

Comment:
Let's use a bigger delay. 1s is too small. And there should be a max.
Let's use delay of 30s and increase 30s for each consecutive failures until max of 2m.
We can use time.Duration directly here.
Something like this: https://go.dev/play/p/LW6tgcfUTd1
Adjust the backoffFactory only. The reconnectDelay and reconnectBackoff should be constant